dygraph/models/unet.py

```diff
@@ -0,0 +1,255 @@
+        loss, probs = fluid.layers.softmax_with_cross_entropy(
+            logit,
+            label,
+            ignore_index=self.ignore_index,
+            return_softmax=True,
+            axis=1)
+
+        loss = loss * mask
+        avg_loss = fluid.layers.mean(loss) / (fluid.layers.mean(mask) + 0.00001)
+
+        label.stop_gradient = True
+        mask.stop_gradient = True
+        return avg_loss
+
+
+class Encoder(fluid.dygraph.Layer):
+    def __init__(self):
+        super().__init__()
+        with scope('encode'):
+            with scope('block1'):
+                self.double_conv = DoubleConv(3, 64)
+            with scope('block1'):
+                self.down1 = Down(64, 128)
+            with scope('block2'):
+                self.down2 = Down(128, 256)
+            with scope('block3'):
+                self.down3 = Down(256, 512)
+            with scope('block4'):
+                self.down4 = Down(512, 512)
+
+    def forward(self, x):
+        short_cuts = []
+        x = self.double_conv(x)
+        short_cuts.append(x)
+        x = self.down1(x)
+        short_cuts.append(x)
+        x = self.down2(x)
+        short_cuts.append(x)
+        x = self.down3(x)
+        short_cuts.append(x)
+        x = self.down4(x)
+        return x, short_cuts
+
+
+class Decode(fluid.dygraph.Layer):
+    def __init__(self, upsample_mode='bilinear'):
+        super().__init__()
+        with scope('decode'):
+            with scope('decode1'):
+                self.up1 = Up(512, 256, upsample_mode)
+            with scope('decode2'):
+                self.up2 = Up(256, 128, upsample_mode)
+            with scope('decode3'):
+                self.up3 = Up(128, 64, upsample_mode)
+            with scope('decode4'):
+                self.up4 = Up(64, 64, upsample_mode)
+
+    def forward(self, x, short_cuts):
+        x = self.up1(x, short_cuts[3])
+        x = self.up2(x, short_cuts[2])
+        x = self.up3(x, short_cuts[1])
+        x = self.up4(x, short_cuts[0])
+        return x
+
+
+class GetLogit(fluid.dygraph.Layer):
+    def __init__(self):
+        super().__init__()
+
+
+class DoubleConv(fluid.dygraph.Layer):
+    def __init__(self, num_channels, num_filters):
+        super().__init__()
+        with scope('conv0'):
+            param_attr = fluid.ParamAttr(
+                name=name_scope + 'weights',
+                regularizer=regularizer,
+                initializer=fluid.initializer.TruncatedNormal(
+                    loc=0.0, scale=0.33))
+            self.conv0 = Conv2D(
+                num_channels=num_channels,
+                num_filters=num_filters,
+                filter_size=3,
+                stride=1,
+                padding=1,
+                param_attr=param_attr)
+            self.bn0 = BatchNorm(
+                num_channels=num_filters,
+                param_attr=fluid.ParamAttr(
+                    name=name_scope + 'gamma', regularizer=regularizer),
```

Comment:
Don't use any `name` for ParamAttr, this is a attr that going to deprecated

Reply:
removed it